core/src/main/java/org/apache/iceberg/BaseTable.java

```diff
@@ -217,4 +218,13 @@ public LocationProvider locationProvider() {
   public String toString() {
     return name();
   }
+
+  Object writeReplace() throws ObjectStreamException {
+    return new StaticTable(ops.io(), name(), ops.current().metadataFileLocation());
+  }
+
+  void init(TableOperations tableOperations, String tableName) {
+    this.ops = tableOperations;
+    this.name = tableName;
+  }
```

Comment:
I don't think that this is needed. Can you remove it?

Reply:
After moving to the `TableStub` solution removed this change